src/testing.py

```diff
@@ -75,7 +75,10 @@ def setlimits():
   def __call__(self, test_file):
     """Execute a single test."""
     basename = os.path.basename(test_file)
-    outfile = self.outname_ctor(self.outdir, test_file) if self.outdir else ''
+    if self.outdir:
+      outfile = self.outname_ctor(self.outdir, test_file, self.extras)
+    else:
+      outfile = None
```

Comment:
I think in this case we want to signal to command_ctor that there is no outfile, passing empty string is actually worse because then someone might try to use it as a filename. 

The only user who currently cares is execute_file.py:execute, and it just check "if outfile" which works either way.  But I think empty string is wrong.